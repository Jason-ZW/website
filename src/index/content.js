'use strict';

import React from 'react';
import { Row, Col } from 'react-bootstrap';

class Section1 extends React.Component {
    displayName: 'Section1';
    constructor() {
        super();
    }
    render() {
        return (
            <div className="section1-container">
                <Row>
                    <Col className="flex-column flex-center" sm={12} md={12} lg={3}>
                        <div className="section1-inner-container">
                            <div className="box-icon">
                                <span>
                                    <i className="fa fa-wrench"/>
                                </span>
                            </div>
                            <div className="box-content">
                                <h4 className="box-title">Honest Services</h4>
                                <div>
                                    Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusa ntium dolore mque.
                                </div>
                            </div>
                        </div>
                    </Col>
                    <Col className="flex-column flex-center" sm={12} md={12} lg={3}>
                        <div className="section1-inner-container">
                            <div className="box-icon" style={{backgroundColor: '#3f51b5'}}>
                                <span>
                                    <i className="fa fa-cog"/>
                                </span>
                            </div>
                            <div className="box-content">
                                <h4 className="box-title">Quick Repair</h4>
                                <div>
                                    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor.
                                </div>
                            </div>
                        </div>
                    </Col>
                    <Col className="flex-column flex-center" sm={12} md={12} lg={3}>
                        <div className="section1-inner-container">
                            <div className="box-icon" style={{backgroundColor: '#673ab7'}}>
                                <span>
                                    <i className="fa fa-history"/>
                                </span>
                            </div>
                            <div className="box-content">
                                <h4 className="box-title">On time delivery</h4>
                                <div>
                                    Sed ut perspiciatis unde omnis iste natus error sit voluptatm accusa ntium dolore mque.
                                </div>
                            </div>
                        </div>
                    </Col>
                    <Col className="flex-column flex-center" sm={12} md={12} lg={3}>
                        <div className="section1-inner-container">
                            <div className="box-icon" style={{backgroundColor: '#ff5722'}}>
                                <span>
                                    <i className="fa fa-heart-o"/>
                                </span>
                            </div>
                            <div className="box-content">
                                <h4 className="box-title">We are pasionate</h4>
                                <div>
                                    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor.
                                </div>
                            </div>
                        </div>
                    </Col>
                </Row>
            </div>
        )
    }
}

module.exports = { Section1 };