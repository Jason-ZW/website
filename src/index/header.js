'use strict';

import React from 'react';
import { Row, Col, Carousel } from 'react-bootstrap';

class TopBar extends React.Component {
    displayName: 'TopBar';
    constructor() {
        super();
    }
    render() {
        return (
            <div className="top-bar">
                <div className="container">
                    <Row>
                        <Col sm={8} md={8}>
                            <div className="top-bar-left">
                                <span>
                                    <i className="fa fa-envelope-o"/> support@think-show.com
                                </span>
                            </div>
                        </Col>
                        <Col className="d-none d-md-block d-sm-block text-right">
                            <div>
                                <span className="top-bar-span-first">
                                    <a className="fa fa-facebook" href="https://www.facebook.com/" title="Facebook" target="_blank"/>
                                </span>
                                <span className="top-bar-span">
                                    <a className="fa fa-google-plus" href="https://plus.google.com/" title="Google" target="_blank"/>
                                </span>
                                <span className="top-bar-span">
                                    <a className="fa fa-twitter" href="https://twitter.com/" title="Twitter" target="_blank"/>
                                </span>
                                <span className="top-bar-span">
                                    <a className="fa fa-linkedin" href="https://www.linkedin.com/" title="Linkedin" target="_blank"/>
                                </span>
                            </div>
                        </Col>
                    </Row>
                </div>
            </div>
        )
    }
}

class Header extends React.Component {
    displayName: 'Header';
    constructor() {
        super();
    }
    render() {
        return (
            <div className="carousel-container">
                <Row className="carousel-row">
                    <Col md={12}>
                        <Carousel>
                            <Carousel.Item>
                                <div className="carousel-mask"/>
                                <div
                                    className="d-block w-100 c-1"
                                />
                                <Carousel.Caption>
                                    <h3>Professional Laptop Repairs </h3>
                                    <p>Specialized in Laptop Service, We working hard to build a reputation of customer satisfaction through technical excellence and friendly staff.</p>
                                    <button type="button" className="btn btn-primary">Learn More</button>
                                </Carousel.Caption>
                            </Carousel.Item>
                            <Carousel.Item>
                                <div className="carousel-mask"/>
                                <div
                                    className="d-block w-100 c-2"
                                />
                                <Carousel.Caption>
                                    <h3>Second slide label</h3>
                                    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
                                    <button type="button" className="btn btn-primary">Learn More</button>
                                </Carousel.Caption>
                            </Carousel.Item>
                            <Carousel.Item>
                                <div className="carousel-mask"/>
                                <div
                                    className="d-block w-100 c-3"
                                />
                                <Carousel.Caption>
                                    <h3>Third slide label</h3>
                                    <p>Praesent commodo cursus magna, vel scelerisque nisl consectetur.</p>
                                    <button type="button" className="btn btn-primary">Learn More</button>
                                </Carousel.Caption>
                            </Carousel.Item>
                        </Carousel>
                    </Col>
                </Row>
            </div>
        )
    }
}

module.exports = { TopBar, Header };