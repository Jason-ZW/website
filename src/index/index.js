'use strict';

import React from 'react';
import ReactDOM from 'react-dom';
import { Row, Col, Carousel, Button } from 'react-bootstrap';
import { TopBar, Header } from './header';
import { Section1, Section2 } from './content';
import 'bootstrap/scss/bootstrap.scss';
import "font-awesome/scss/font-awesome.scss";
import '../../static/css/index.scss';

class EntryPoint extends React.Component {
    displayName: 'EntryPoint';
    render() {
        return (
            <div>
                <TopBar/>
                <Header/>
                <Section1/>
                <Section2/>
            </div>
        )
    }
}

ReactDOM.render(
    <EntryPoint/>,
    document.getElementById('root')
);