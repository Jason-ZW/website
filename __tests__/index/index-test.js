'use strict';
import expect from 'expect';
import React from 'react';
import { TopBar, Header } from '../../src/index/header';
import { Section1 } from '../../src/index/content';
import renderer from 'react-test-renderer';

describe('renders correctly', () => {
    it('topbar correctly', () => {
        const tree = renderer.create(<TopBar />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('header correctly', () => {
        const tree = renderer.create(<Header />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('section1 correctly', () => {
        const tree = renderer.create(<Section1 />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});


