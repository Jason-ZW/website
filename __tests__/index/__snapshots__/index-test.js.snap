// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly header correctly 1`] = `
<div
  className="carousel-container"
>
  <div
    className="carousel-row row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="carousel slide"
        onKeyDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <ol
          className="carousel-indicators"
        >
          <li
            className="active"
            onClick={[Function]}
          />
           
          <li
            className={null}
            onClick={[Function]}
          />
           
          <li
            className={null}
            onClick={[Function]}
          />
           
        </ol>
        <div
          className="carousel-inner"
        >
          <div
            className="carousel-item active carousel-item"
          >
            <div
              className="carousel-mask"
            />
            <div
              className="d-block w-100 c-1"
            />
            <div
              className="carousel-caption"
            >
              <h3>
                Professional Laptop Repairs 
              </h3>
              <p>
                Specialized in Laptop Service, We working hard to build a reputation of customer satisfaction through technical excellence and friendly staff.
              </p>
              <button
                className="btn btn-primary"
                type="button"
              >
                Learn More
              </button>
            </div>
          </div>
          <div
            className="carousel-item carousel-item"
          >
            <div
              className="carousel-mask"
            />
            <div
              className="d-block w-100 c-2"
            />
            <div
              className="carousel-caption"
            >
              <h3>
                Second slide label
              </h3>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit.
              </p>
              <button
                className="btn btn-primary"
                type="button"
              >
                Learn More
              </button>
            </div>
          </div>
          <div
            className="carousel-item carousel-item"
          >
            <div
              className="carousel-mask"
            />
            <div
              className="d-block w-100 c-3"
            />
            <div
              className="carousel-caption"
            >
              <h3>
                Third slide label
              </h3>
              <p>
                Praesent commodo cursus magna, vel scelerisque nisl consectetur.
              </p>
              <button
                className="btn btn-primary"
                type="button"
              >
                Learn More
              </button>
            </div>
          </div>
        </div>
        <a
          className="carousel-control-prev"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <span
            aria-hidden="true"
            className="carousel-control-prev-icon"
          />
          <span
            className="sr-only"
          >
            Previous
          </span>
        </a>
        <a
          className="carousel-control-next"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <span
            aria-hidden="true"
            className="carousel-control-next-icon"
          />
          <span
            className="sr-only"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly section1 correctly 1`] = `
<div
  className="section1-container"
>
  <div
    className="row"
  >
    <div
      className="flex-column flex-center col-lg-3 col-md-12 col-sm-12"
    >
      <div
        className="section1-inner-container"
      >
        <div
          className="box-icon"
        >
          <span>
            <i
              className="fa fa-wrench"
            />
          </span>
        </div>
        <div
          className="box-content"
        >
          <h4
            className="box-title"
          >
            Honest Services
          </h4>
          <div>
            Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusa ntium dolore mque.
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex-column flex-center col-lg-3 col-md-12 col-sm-12"
    >
      <div
        className="section1-inner-container"
      >
        <div
          className="box-icon"
          style={
            Object {
              "backgroundColor": "#3f51b5",
            }
          }
        >
          <span>
            <i
              className="fa fa-cog"
            />
          </span>
        </div>
        <div
          className="box-content"
        >
          <h4
            className="box-title"
          >
            Quick Repair
          </h4>
          <div>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor.
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex-column flex-center col-lg-3 col-md-12 col-sm-12"
    >
      <div
        className="section1-inner-container"
      >
        <div
          className="box-icon"
          style={
            Object {
              "backgroundColor": "#673ab7",
            }
          }
        >
          <span>
            <i
              className="fa fa-history"
            />
          </span>
        </div>
        <div
          className="box-content"
        >
          <h4
            className="box-title"
          >
            On time delivery
          </h4>
          <div>
            Sed ut perspiciatis unde omnis iste natus error sit voluptatm accusa ntium dolore mque.
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex-column flex-center col-lg-3 col-md-12 col-sm-12"
    >
      <div
        className="section1-inner-container"
      >
        <div
          className="box-icon"
          style={
            Object {
              "backgroundColor": "#ff5722",
            }
          }
        >
          <span>
            <i
              className="fa fa-heart-o"
            />
          </span>
        </div>
        <div
          className="box-content"
        >
          <h4
            className="box-title"
          >
            We are pasionate
          </h4>
          <div>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly topbar correctly 1`] = `
<div
  className="top-bar"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-8 col-sm-8"
      >
        <div
          className="top-bar-left"
        >
          <span>
            <i
              className="fa fa-envelope-o"
            />
             supporter@3-nice.com
          </span>
        </div>
      </div>
      <div
        className="d-none d-md-block d-sm-block text-right col"
      >
        <div>
          <span
            className="top-bar-span-first"
          >
            <a
              className="fa fa-facebook"
              href="https://www.facebook.com/"
              target="_blank"
              title="Facebook"
            />
          </span>
          <span
            className="top-bar-span"
          >
            <a
              className="fa fa-google-plus"
              href="https://plus.google.com/"
              target="_blank"
              title="Google"
            />
          </span>
          <span
            className="top-bar-span"
          >
            <a
              className="fa fa-twitter"
              href="https://twitter.com/"
              target="_blank"
              title="Twitter"
            />
          </span>
          <span
            className="top-bar-span"
          >
            <a
              className="fa fa-linkedin"
              href="https://www.linkedin.com/"
              target="_blank"
              title="Linkedin"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
